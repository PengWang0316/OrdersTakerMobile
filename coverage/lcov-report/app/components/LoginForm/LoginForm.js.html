<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/LoginForm/LoginForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/LoginForm</a> LoginForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.31% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>48/52</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>23/25</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">26x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">24x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component, Fragment } from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';
import { Input, Icon, Button } from 'react-native-elements';
&nbsp;
import { loginWithPassword, emptyUser } from '../../actions/UserActions';
import styles from './Styles';
&nbsp;
type Props = {
  user: Object,
  loginWithPassword: Function,
  emptyUser: Function,
  handleSnackbarUpdate: Function,
};
&nbsp;
type States = {
  username: string,
  password: string,
  isReady: boolean,
  isError: boolean,
  loginWithPassword: Function,
  handleSnackbarUpdate: Function,
  user: Object,
};
&nbsp;
const LOGIN_FAIL_MESSAGE = 'Wrong username or password';
const COLOR_ERROR = '#ff1744';
&nbsp;
/**
 * The login form component.
 */
export class LoginForm extends Component&lt;Props, States&gt; {
  state = {
    username: '',
    password: '',
    isReady: false,
    isSubmitted: false,
  };
&nbsp;
  /**
   * Call the onToggleSnackbar when isSubmitted state is true and props received a null for user.
   * @param {object} nextProps contains the new props value.
   * @param {object} prevState contains the previous state's value.
   * @return {Object} Return an object to change the state.
   */
  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.isSubmitted &amp;&amp; nextProps.user.isFail) {
      nextProps.handleSnackbarUpdate({
        isShowSnackbar: true, snackbarMessage: LOGIN_FAIL_MESSAGE, snackbarBgColor: COLOR_ERROR,
      });
      nextProps.emptyUser(); // Reset user state to an empty object. So, next time the back-end code still can send a isFail mark when the authentication fails.
      return { isSubmitted: false };
    }
    return null;
  }
&nbsp;
  /**
   * Call the backend to validate login information.
   * @return {null} No return.
   */
  handleLoginClick = () =&gt; this.setState(
    { isSubmitted: true },
    () =&gt; this.props.loginWithPassword({
      username: this.state.username,
      password: this.state.password,
    }),
  );
&nbsp;
  /**
   * Update the username and isReady state when a user type in the text input.
   * @param {string} text is the string a user is typing.
   * @return {null} No return.
   */
  handleUsernameTextChange = text =&gt; this.setState(({ password }) =&gt; ({
    username: text,
    isReady: text.length !== 0 &amp;&amp; password.length !== 0,
  }));
&nbsp;
  /**
   * Update the password and isReady state when a user type in the text input.
   * @param {string} text is the string a user is typing.
   * @return {null} No return.
   */
  handlePasswordTextChange = text =&gt; this.setState(({ username }) =&gt; ({
    password: text,
    isReady: text.length !== 0 &amp;&amp; username.length !== 0,
  }));
&nbsp;
  /**
   * The render method for the component.
   * @return {jsx} Return jsx for the component.
   */
  render() {
    const {
      username, password, isReady, isSubmitted,
    } = this.state;
&nbsp;
    return (
      &lt;Fragment&gt;
        &lt;View&gt;
          &lt;Text style={styles.title}&gt;Username Login&lt;/Text&gt;
        &lt;/View&gt;
&nbsp;
        &lt;View&gt;
          &lt;Input
            onChangeText={this.handleUsernameTextChange}
            placeholder="Username"
            value={username}
            leftIcon={(
              &lt;Icon
                type="font-awesome"
                name="user-o"
                size={18}
                color="gray"
              /&gt;
            )}
          /&gt;
        &lt;/View&gt;
&nbsp;
        &lt;View&gt;
          &lt;Input
            onChangeText={this.handlePasswordTextChange}
            value={password}
            secureTextEntry
            placeholder="Password"
            leftIcon={(
              &lt;Icon
                type="feather"
                name="lock"
                size={18}
                color="gray"
              /&gt;
            )}
          /&gt;
        &lt;/View&gt;
&nbsp;
        &lt;View style={styles.buttonView}&gt;
          &lt;Button
            disabled={isSubmitted || !isReady}
            title="LOGIN"
            onPress={this.handleLoginClick}
            buttonStyle={styles.loginBtn}
          /&gt;
        &lt;/View&gt;
&nbsp;
      &lt;/Fragment&gt;
    );
  }
}
/* istanbul ignore next */
const mapStateToProps = state =<span class="fstat-no" title="function not covered" >&gt; ({ user: state</span>.user })<span class="cstat-no" title="statement not covered" >;</span>
&nbsp;
/* istanbul ignore next */
const mapDispatchToProprs = dispatc<span class="fstat-no" title="function not covered" >h =&gt; ({<span class="cstat-no" title="statement not covered" ></span></span>
  loginWithPassword: params =&gt; d<span class="fstat-no" title="function not covered" >ispatch(loginWithP</span>assword(p<span class="cstat-no" title="statement not covered" >arams)),</span>
  emptyUser: () =&gt; dispa<span class="fstat-no" title="function not covered" >tch(emptyU</span>ser<span class="cstat-no" title="statement not covered" >()),</span>
});
export default connect(mapStateToProps, mapDispatchToProprs)(LoginForm);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Sep 27 2018 22:54:50 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
