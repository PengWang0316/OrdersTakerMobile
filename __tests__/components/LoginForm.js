import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { LoginForm } from '../../app/components/LoginForm/LoginForm';

jest.mock('Text', () => 'Text');
jest.mock('View', () => 'View');
// jest.mock('Input', () => 'Input');
// jest.mock('Icon', () => 'Icon');
// jest.mock('Button', () => 'Button');

describe('LoginForm', () => {
  const defaultProps = {
    user: {},
    handleSnackbarUpdate: jest.fn(),
    emptyUser: jest.fn(),
    loginWithPassword: jest.fn(),
  };
  const getShallowComponent = (props = defaultProps) => shallow(<LoginForm {...props} />);

  test('Initial states', () => {
    const component = getShallowComponent();
    const {
      username, password, isReady, isSubmitted,
    } = component.instance().state;

    expect(username).toBe('');
    expect(password).toBe('');
    expect(isReady).toBe(false);
    expect(isSubmitted).toBe(false);
  });

  test('getDerivedStateFromProps isSubmitted false', () => expect(LoginForm.getDerivedStateFromProps(null, { isSubmitted: false })).toBeNull());
  test('getDerivedStateFromProps isSubmitted is true and user.isFail is false', () => expect(LoginForm.getDerivedStateFromProps({ user: { isFail: false } }, { isSubmitted: true })).toBeNull());
  test('getDerivedStateFromProps isSubmitted is true and user.isFail is true', () => {
    const mockUpdateFn = jest.fn();
    const mockEmptyUserFn = jest.fn();
    const nextProps = {
      user: { isFail: true },
      handleSnackbarUpdate: mockUpdateFn,
      emptyUser: mockEmptyUserFn,
    };
    const prevState = { isSubmitted: true };
    expect(LoginForm.getDerivedStateFromProps(nextProps, prevState)).toEqual({ isSubmitted: false })
    expect(mockUpdateFn).toHaveBeenCalledTimes(1);
    expect(mockUpdateFn).toHaveBeenLastCalledWith({ isShowSnackbar: true, snackbarMessage: 'Wrong username or password', snackbarBgColor: '#ff1744' });
    expect(mockEmptyUserFn).toHaveBeenCalledTimes(1);
  });

  test('handleLoginClick', () => {
    const component = getShallowComponent();
    const someStates = { username: 'username', password: 'password' };
    component.setState(someStates);
    component.instance().handleLoginClick();
    expect(component.state('isSubmitted')).toBe(true);
    expect(defaultProps.loginWithPassword).toHaveBeenCalledTimes(1);
    expect(defaultProps.loginWithPassword).toHaveBeenLastCalledWith(someStates);
  });

  test('handleUsernameTextChange text length equal 0', () => {
    const component = getShallowComponent();
    const text = '';
    component.instance().handleUsernameTextChange(text);
    expect(component.state('username')).toBe(text);
    expect(component.state('isReady')).toBe(false);
  });

  test('handleUsernameTextChange passowrd length equal 0', () => {
    const component = getShallowComponent();
    const text = 'text';
    component.setState({ password: '' });
    component.instance().handleUsernameTextChange(text);
    expect(component.state('username')).toBe(text);
    expect(component.state('isReady')).toBe(false);
  });

  test('handleUsernameTextChange username and password length not equal 0', () => {
    const component = getShallowComponent();
    const text = 'text';
    component.setState({ password: 'password' });
    component.instance().handleUsernameTextChange(text);
    expect(component.state('username')).toBe(text);
    expect(component.state('isReady')).toBe(true);
  });

  test('handlePasswordTextChange password length equal 0', () => {
    const component = getShallowComponent();
    const password = '';
    component.instance().handlePasswordTextChange(password);
    expect(component.state('password')).toBe(password);
    expect(component.state('isReady')).toBe(false);
  });

  test('handlePasswordTextChange username length equal 0', () => {
    const component = getShallowComponent();
    const password = 'password';
    component.setState({ username: '' });
    component.instance().handlePasswordTextChange(password);
    expect(component.state('password')).toBe(password);
    expect(component.state('isReady')).toBe(false);
  });

  test('handlePasswordTextChange username and password length not equal 0', () => {
    const component = getShallowComponent();
    const password = 'password';
    component.setState({ username: 'username' });
    component.instance().handlePasswordTextChange(password);
    expect(component.state('password')).toBe(password);
    expect(component.state('isReady')).toBe(true);
  });

  test('Snapshot', () => expect(renderer.create(<LoginForm {...defaultProps} />).toJSON()).toMatchSnapshot());
});
