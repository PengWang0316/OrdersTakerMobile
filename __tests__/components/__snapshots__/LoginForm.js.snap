// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Snapshot 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "color": "#6d9cb5",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 15,
        }
      }
    >
      Username Login
    </Text>
  </View>,
  <View>
    <View
      style={
        Array [
          Object {
            "width": "90%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "marginLeft": 15,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                ]
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "height": 40,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>,
  <View>
    <View
      style={
        Array [
          Object {
            "width": "90%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "marginLeft": 15,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                ]
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "height": 40,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          false,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderRadius": 3,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#6d9cb5",
                "borderRadius": 15,
              },
              Object {
                "backgroundColor": "#D1D5D8",
              },
              undefined,
              false,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "color": "#F3F4F5",
                },
                undefined,
              ]
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
